src/main/java/com/auth0/net/Telemetry.java

```diff
@@ -4,50 +4,81 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import org.apache.commons.codec.binary.Base64;
 
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
 @SuppressWarnings("WeakerAccess")
-class Telemetry {
+public class Telemetry {
     static final String HEADER_NAME = "Auth0-Client";
 
     private static final String NAME_KEY = "name";
     private static final String VERSION_KEY = "version";
+    private static final String CORE_KEY = "core";
+    private static final String ENV_KEY = "env";
+    private static final String JAVA_KEY = "java";
 
     private final String name;
     private final String version;
+    private final String core;
+    private final Map<String, String> env;
+    private final String value;
 
-    Telemetry(String name, String version) {
-        this.name = name;
-        this.version = version;
-    }
-
-    public String getName() {
-        return name;
+    public Telemetry(String name, String version) {
+        this(name, version, null);
     }
 
-    public String getVersion() {
-        return version;
-    }
+    public Telemetry(String name, String version, String core) {
+        this.name = name;
+        this.version = version;
+        this.core = core;
```

Comment:
What is `core` here? If it's used to extend this SDK's telemetry, let's put the SDK name instead to be more clear.

Reply:
`core` for this library is nothing (null). But those libs that use this internally, would expose as `core` the version of auth0-java (this sdk). In other SDKs we named it "libraryVersion", I can change to that if you agree. Remember though `env` includes `core` as per the RFC. 

e.g. on android https://github.com/auth0/Lock.Android/blob/eeff3594786520d50ba3938fc9f7ebf2f80588b8/lib/src/main/java/com/auth0/android/lock/Lock.java#L232

Reply:
RFC mentioned `core` long ago (mostly in the context of libs using auth0.js) but it's not there now. Let's use the library name itself, that will be much clearer when pulling reports. I'l add that to the RFC now.